lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp

```diff
@@ -304,20 +304,28 @@ class BlockedToMMA : public mlir::RewritePattern {
     } else {
 
       // convert operands
-      int minBitwidth =
-          std::min(computeOrigBitWidth(a), computeOrigBitWidth(b));
-      Type minType = IntegerType::get(ctx, minBitwidth);
+      int aBitWidth = computeOrigBitWidth(a);
+      int bBitWidth = computeOrigBitWidth(b);
+      int minBitwidth = std::min(aBitWidth, bBitWidth);
+      // TODO: remove s8xf32 matmul special case when issue is fixed.
+      // https://github.com/openai/triton/issues/2853
+      if (minBitwidth == 8 && std::max(aBitWidth, bBitWidth) == 32) {
+        minBitwidth = 0;
+      }
+      Type aEltType = oldAType.getElementType();
+      Type bEltType = oldBType.getElementType();
+      if (minBitwidth > 0) {
+        aEltType = bEltType = IntegerType::get(ctx, minBitwidth);
+      }
```

Comment:
Could this be written as

```
if (minBitwidth != 8 || max(aBitWidth, bBitWidth) != 32) {
  aEltType = bEltType = IntegerType::get(ctx, minBitWidth);
}
```

?

Reply:
Or `if (minBitWidth == 8 && max(aBitWidth, bBitWidth) == 32) return failure()`, if that's what you're meaning to do?

Reply:
Done

Reply:
No, s8xf32 dots do not produce wrong result now. I apologize for the confusion, let me try to explain it one more time.

Currently it always crashes with segfault for all s8xf32 dots.
This patch + [disabled pipelining] makes the compiler work correctly and produce the correct results for s8xf32 dots.
This patch + [pipelining] still crashes with segfault later on.

Reply:
> why do you think it can produce wrong answer?

Because the comment says

> // This doesn't allow the enablement of s8xf32 dots yet

I read this as saying that s8xf32 dots cannot be enabled because they're not correct yet.

I think you mean we cannot enable *pipelining* for s8xf32 dots?

Sorry to be a pain, Aliia, but the comment needs clarification before I can accept this patch.

- Instead of saying "Remove s8xf32 matmul special case when the issue is fixed:", first we should explain what the special case is doing.  "Disable matmul pipelining for s8xf32 matmuls"?
- Then explain how this branch disables pipelining.  I don't understand how the code does this, so I can't suggest verbiage.
- Then explain why we're doing it.  "There is a bug that we don't yet understand in the pipeliner where if we pipeline s8xf32 matmuls, the pipeliner segfaults, https://github.com/openai/triton/issues/2853"
- Finally add the TODO.  "TODO: Remove the s8xf32 check once we fix the pipeliner bug."